wpimath/src/main/native/include/frc/interpolatable/TimeInterpolatableBuffer.h

```diff
@@ -0,0 +1,78 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <array>
+#include <utility>
+#include <vector>
+
+#include <wpi/MathExtras.h>
+
+#include "units/time.h"
+
+namespace frc {
+
+template <typename T>
+class TimeInterpolatableBuffer {
```

Comment:
I think we should also support custom interpolation functions besides `wpi::Lerp`. Maybe an optional constructor parameter?

```
explicit TimeInterpolatableBuffer(std::function<T(const T&, const T&, double)> func
  = [](const T& start, const T& end, double t) { return wpi::Lerp(start, end, t); }) :
  m_interpolatingFunc(func) {}
```

Reply:
Addressed I think